### PR TITLE
Client refactor

### PR DESCRIPTION
**Pull request by jamesmuerle**
Removed the ajax call for the user list and added the logic to joining and hosting the game. Also removed the populatePlayerList method from GameState, since it wasn't really a GameState method.

### CODE REVIEW

File: game/static/game/js/renderer.js [hunk 1]
```diff
@@ -21,14 +23,17 @@ function Canvas(App) {
     self.ctx.font = "14px Helvetica";
     self.canvas.tabIndex = "0";
         
+    // Generates a random hex color
     self.randomColor = function() {
```
Comment (hellochar) [inline game/static/game/js/renderer.js]: I would suggest defining instance methods in javascript by putting it in the prototype, like

``` javascript
function Renderer() {
    ...
}

Renderer.prototype.randomColor = function() {
    ....
}
```

File: game/static/game/js/renderer.js [hunk 2]
```diff
@@ -36,56 +41,71 @@ function Canvas(App) {
...
+
+        self.ctx.fillStyle = "Black";
+
+        if (game.conn_state == game.GAME_STATES.CONNECTED) {
             // For each player in the gamestate
-            var players = App.gamestate.players;
+            var players = game.gamestate.players;
             for (var playerind in players) {
                 if (players.hasOwnProperty(playerind)) {
                     var player = players[playerind];
```
Comment (hellochar) [inline game/static/game/js/renderer.js]: You can replace the last three lines with

``` javascript
players.forEach(function (player, playerind) {
....
}
```

File: game/static/game/js/renderer.js [hunk 3]
```diff
@@ -36,56 +41,71 @@ function Canvas(App) {
...
                     // For each unit in the player
                     for (var unitind in player.units) {
                         if (player.units.hasOwnProperty(unitind)) {
-                            var unit = player.units[unitind];
-                            var pos = unit.pos(nowtime - App.client_start_time);
-                            var color = App.gamestate.colors[playerind];
-                            self.drawCircle(pos.x, pos.y, 20, color);
+                            self.renderUnit(
+                                player.units[unitind],
```
Comment (hellochar) [inline game/static/game/js/renderer.js]: same here; you can just do

``` javascript
player.units.forEach(function(unit) { self.renderUnit(unit....); })
```

**Comment by hellochar** [comment]
looks good, but there are merge conflicts? also lets get some tests for game/views.py methods since they look pretty logic intensive

**Comment by hellochar** [comment]
I just merged #11 (map editor integration) which touches a lot of the same files; lets merge that shit @strayhobo @flawedmatrix

File: game/static/game/js/renderer.js (hunk 1 above)
Comment (flawedmatrix) [inline game/static/game/js/renderer.js]: In this particular case it's probably better to define this as something equivalent to a static method.

**Comment by jamesmuerle** [comment]
Looks good. A lot was rewritten here, so for next time let's try to separate it into different commits.

**Pull request merged**